local fishes = {
    salmon = {
        name = "Salmon",
        weight = 15, -- Standard weight in kg or lbs
        value = 50, -- How much it sells for
        rarity = "Common",
        image = "rbxassetid://1234567890",
    },
    
    golden_shark = {
        name = "Golden Shark",
        weight = 2e6, -- Using scientific notation for a massive/legendary fish (2,000,000)
        value = 15000,
        rarity = "Legendary",
        image = "rbxassetid://0987654321",
    },
}

return fishes